emily/handler/tests/integration/endpoints/deposit.rs

```diff
@@ -339,3 +352,157 @@ async fn get_failed_deposits() {
     assert_eq!(response.deposits.len(), 0);
     client.teardown().await;
 }
+
+/// Update deposits test.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn update_deposit() {
+    // Arrange.
+    let client = setup_deposit_integration_test().await;
+
+    // Get a random deposit info and its identifying fields.
+    let deposit_keys: Vec<DepositEntryKey> = client
+        .get_all_deposits()
+        .await
+        .into_iter()
+        .map(|deposit_info| DepositEntryKey {
+            bitcoin_tx_output_index: deposit_info.bitcoin_tx_output_index,
+            bitcoin_txid: deposit_info.bitcoin_txid,
+        })
+        .collect();
+
+    let key1 = deposit_keys.get(0).unwrap().clone();
+    let key2 = deposit_keys.get(1).unwrap().clone();
+
+    let bitcoin_txid = key1.bitcoin_txid.clone();
+    let bitcoin_tx_output_index = key1.bitcoin_tx_output_index;
+
+    // Get the full deposit.
+    let _original_deposit = client
+        .get_deposit(&bitcoin_txid, bitcoin_tx_output_index)
+        .await;
+
+    // Make some parameters.
+    let updated_hash = "UPDATED_HASH".to_string();
+    let updated_height: u64 = 12345;
+    let updated_status: Status = Status::Accepted;
+    let updated_message: String = "UPDATED_MESSAGE".to_string();
+    let fulfillment: Fulfillment = Fulfillment {
+        bitcoin_txid: "FULFILLMENT_BITCOIN_TXID".to_string(),
+        bitcoin_tx_index: 10,
+        stacks_txid: "FULFILLMENT_STACKS_TXID".to_string(),
+        bitcoin_block_hash: "FULFILLMENT_HASH".to_string(),
+        bitcoin_block_height: 10,
+        btc_fee: 12,
+    };
+
+    // Create and make the request.
+    let single_update = UpdateDepositsRequestBody {
+        deposits: vec![
+            DepositUpdate {
+                // Original fields.
+                bitcoin_txid: key1.bitcoin_txid.clone(),
+                bitcoin_tx_output_index: key1.bitcoin_tx_output_index,
+                // New updated height.
+                last_update_height: updated_height,
+                last_update_block_hash: updated_hash.clone(),
+                status: updated_status.clone(),
+                status_message: updated_message.clone(),
+                fulfillment: Some(fulfillment.clone()),
+            },
+            DepositUpdate {
+                // Original fields.
+                bitcoin_txid: key2.bitcoin_txid.clone(),
+                bitcoin_tx_output_index: key2.bitcoin_tx_output_index,
+                // New updated height.
+                last_update_height: updated_height,
+                last_update_block_hash: updated_hash.clone(),
+                status: updated_status.clone(),
+                status_message: updated_message.clone(),
+                fulfillment: Some(fulfillment.clone()),
+            },
+        ],
+    };
+    let response = client.update_deposits(&single_update).await;
+    assert_eq!(response.deposits.len(), 2);
```

Comment:
I would change checks like this to `assert_eq!(response.deposits.len(), single_update.deposits.len());`

Reply:
Done!